### PR TITLE
fix: fixed tests in test_api.py

### PR DESCRIPTION
**Pull request by linglp**
## Chagelog:  
* Broken downn tests related to `test_submit_manifest`, so that it is easier to track what tests failed
* Since DCA and DFA are using csv file to submit instead of json str, I modified the tests so that the majority of the tests focused on testing using the CSV instead of json. 
* Modified the github workflow so that the API tests (except those related to benchmarking performance) get run when pushing to develop or main branch

## To test this PR
Run all the tests in test_api.py locally. All the tests should pass except some in TestValidationBenchmark

Related to Jira issue: https://sagebionetworks.jira.com/jira/software/c/projects/FDS/issues/FDS-318?jql=project%20%3D%20%22FDS%22%20AND%20text%20~%20%22test%22%20AND%20status%20IN%20%28Open%2C%22Ready%20for%20Dev%20Triage%22%2C%22Ready%20for%20Team%22%29%20ORDER%20BY%20created%20DESC

### CODE REVIEW

File: tests/test_api.py
Comment (linglp) [inline tests/test_api.py:40]: Note: I added the test manifest using biospecimen component to make sure that it won't mess up with cross validation rules.

File: .github/workflows/api_test.yml
Comment (linglp) [inline .github/workflows/api_test.yml:8]: @linglp trigger the run when a PR is requested for review.

File: tests/test_api.py
Comment (linglp) [inline tests/test_api.py:52]: Look into why `rdb_table_manifest.csv` has error when upserting manifest

**Review by GiaJordan — CHANGES_REQUESTED** [review]

File: .github/workflows/api_test.yml
Comment (linglp) [reply .github/workflows/api_test.yml]: Yes. I have made that change. But it is a bit difficult to test on my own since my test repo does not have a collaborator. Can I add you as a collaborator so that i could test out the workflow before merging?

Comment (linglp) [reply .github/workflows/api_test.yml]: hmm maybe this is not needed. I could see in github action that the tests are running.

Comment (GiaJordan) [reply .github/workflows/api_test.yml]: I can see them running from my end as well

Comment (GiaJordan) [reply .github/workflows/api_test.yml]: @linglp Just a warning that the benchmarking tests will likely fail as everything stands right now. 
@milen-sage we should decide what our acceptable time is for the validation rules as things currently stand and set the tests to ensure execution time is below that, and then when we make the optimizations decrease the time to what we're targeting, otherwise the api tests will always show as failing and prevent merging without an override.

Comment (linglp) [reply .github/workflows/api_test.yml]: I noticed that if I re-requested review. The tests don't get run again.

**Comment by linglp** [comment]
@GiaJordan @mialy-defelice I think running `test_api.py` when a review gets requested might not be a good idea. When I experimented, it seems like if you re-request a review from a reviewer, `pull_request_review` doesn't really trigger the workflow run, and you have to remove the reviewer and add the reviewer back to get the workflow triggered. Also, if this workflow gets run concurrently on Github action, synapse is likely to raise a 403 error likes we have seen before: 
```
ERROR    schematic_api.api:app.py:1449 Exception on /v1/model/submit [POST]
Traceback (most recent call last):
  File "/home/runner/work/schematic/schematic/.venv/lib/python3.9/site-packages/synapseclient/client.py", line 3984, in _handle_synapse_http_error
    exceptions._raise_for_status(response, verbose=self.debug)
  File "/home/runner/work/schematic/schematic/.venv/lib/python3.9/site-packages/synapseclient/core/exceptions.py", line 160, in _raise_for_status
    raise SynapseHTTPError(message, response=response)
synapseclient.core.exceptions.SynapseHTTPError: 403 Client Error: 
You lack UPDATE access to the requested entity.
```

**Comment by mialy-defelice** [comment]
A bunch of tests are failing for me. I tried again and it timed out. These are the ones I have failing so far.
```
FAILED tests/test_api.py::TestManifestOperation::test_manifest_download[None-Example.csv-syn51078535-BulkRNA-seqAssay-synapse_storage_manifest.csv] - assert 500 == 200
FAILED tests/test_api.py::TestManifestOperation::test_manifest_download[None-Example.csv-syn51156998-Biospecimen-synapse_storage_manifest_biospecimen.csv] - assert 500 == 200
FAILED tests/test_api.py::TestManifestOperation::test_manifest_download[True-None-syn51078535-BulkRNA-seqAssay-synapse_storage_manifest.csv] - assert 500 == 200
FAILED tests/test_api.py::TestManifestOperation::test_manifest_download[True-None-syn51156998-Biospecimen-synapse_storage_manifest_biospecimen.csv] - assert 500 == 200
FAILED tests/test_api.py::TestManifestOperation::test_manifest_download[True-Example.csv-syn51078535-BulkRNA-seqAssay-synapse_storage_manifest.csv] - assert 500 == 200
FAILED tests/test_api.py::TestManifestOperation::test_manifest_download[True-Example.csv-syn51156998-Biospecimen-synapse_storage_manifest_biospecimen.csv] - assert 500 == 200
FAILED tests/test_api.py::TestManifestOperation::test_manifest_download[False-None-syn51078535-BulkRNA-seqAssay-synapse_storage_manifest.csv] - assert 500 == 200
FAILED tests/test_api.py::TestManifestOperation::test_manifest_download[False-None-syn51156998-Biospecimen-synapse_storage_manifest_biospecimen.csv] - assert 500 == 200
FAILED tests/test_api.py::TestManifestOperation::test_manifest_download[False-Example.csv-syn51078535-BulkRNA-seqAssay-synapse_storage_manifest.csv] - assert 500 == 200
FAILED tests/test_api.py::TestManifestOperation::test_manifest_download[False-Example.csv-syn51156998-Biospecimen-synapse_storage_manifest_biospecimen.csv] - assert 500 == 200
FAILED tests/test_api.py::TestManifestOperation::test_dataset_manifest_download[None-None] - assert 500 == 200
FAILED tests/test_api.py::TestManifestOperation::test_dataset_manifest_download[None-True] - assert 500 == 200
FAILED tests/test_api.py::TestManifestOperation::test_dataset_manifest_download[None-False] - assert 500 == 200
FAILED tests/test_api.py::TestManifestOperation::test_dataset_manifest_download[Test-None] - assert 500 == 200
FAILED tests/test_api.py::TestManifestOperation::test_dataset_manifest_download[Test-True] - assert 500 == 200
FAILED tests/test_api.py::TestManifestOperation::test_submit_manifest_table_and_file_replace - assert 500 == 200
FAILED tests/test_api.py::TestManifestOperation::test_submit_manifest_file_only_replace - assert 500 == 200
FAILED tests/test_api.py::TestManifestOperation::test_submit_manifest_json_str_replace - assert 500 == 200
FAILED tests/test_api.py::TestManifestOperation::test_submit_manifest_table_and_file_upsert - assert 500 == 200
```

**Comment by mialy-defelice** [comment]
Looks like the same tests are failing GitHub Actions.

**Comment by mialy-defelice** [comment]
> @GiaJordan @mialy-defelice I think running `test_api.py` when a review gets requested might not be a good idea. When I experimented, it seems like if you re-request a review from a reviewer, `pull_request_review` doesn't really trigger the workflow run, and you have to remove the reviewer and add the reviewer back to get the workflow triggered. Also, if this workflow gets run concurrently on Github action, synapse is likely to raise a 403 error likes we have seen before:
> 
> ```
> ERROR    schematic_api.api:app.py:1449 Exception on /v1/model/submit [POST]
> Traceback (most recent call last):
>   File "/home/runner/work/schematic/schematic/.venv/lib/python3.9/site-packages/synapseclient/client.py", line 3984, in _handle_synapse_http_error
>     exceptions._raise_for_status(response, verbose=self.debug)
>   File "/home/runner/work/schematic/schematic/.venv/lib/python3.9/site-packages/synapseclient/core/exceptions.py", line 160, in _raise_for_status
>     raise SynapseHTTPError(message, response=response)
> synapseclient.core.exceptions.SynapseHTTPError: 403 Client Error: 
> You lack UPDATE access to the requested entity.
> ```

Let's chat about this in code review.

**Comment by linglp** [comment]
As @mialy-defelice and i discussed in the code review, I will keep this PR as fixing the tests for now. We could decide later what we want to do with running test_api.py remotely on Github.

**Review by mialy-defelice — APPROVED** [review]

**Comment by linglp** [comment]
@GiaJordan If you don't have time to test again, could you also approve this PR? We are looking to include this PR in the upcoming release.

**Pull request merged**
